game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -45,8 +46,27 @@ class ResourceType(grid:Grid, val min: Int, val max: Int, mass: Int, score: Int)
 
   def spawnResources: Unit = {
     val ratio = (positions.length - min).toFloat / (max - min)
-    if (scala.util.Random.nextFloat >= ratio) {
+    if ((scala.util.Random.nextFloat + 0.000001) >= ratio) {
       positions :::= List(grid.randomPosition)
     }
   }
-}
\ No newline at end of file
+
+  /** Small loop to detect if a cell can eat a resource **/
+  def detectCollision(players: List[Player]): Unit = {
+    for(player <- players) {
+      for(cell <- player.cells) {
+        for(position <- positions){
```

Comment:
I agree, we could do space partitioning here (probably using an existing lib rather than doing a custom one), if we ever need to (if profiling shows this a bottleneck)!